src/dotnet-new3/BuiltInTemplatePackagesProvider.cs

```diff
@@ -51,12 +52,18 @@ public BuiltInTemplatePackagesProvider(BuiltInTemplatePackagesProviderFactory fa
             public Task<IReadOnlyList<ITemplatePackage>> GetAllTemplatePackagesAsync(CancellationToken cancellationToken)
             {
                 List<ITemplatePackage> templatePackages = new List<ITemplatePackage>();
-                string assemblyLocation = Path.GetDirectoryName(typeof(Program).Assembly.Location);
+                string? assemblyLocation = Path.GetDirectoryName(typeof(Program).Assembly.Location);
                 string dn3Path = _settings.Environment.GetEnvironmentVariable("DN3");
 
+                if (string.IsNullOrWhiteSpace(assemblyLocation))
+                {
+                    _settings.Host.LogDiagnosticMessage("Couldn't the setup built-in packages.", "Install");
+                    return Task.FromResult((IReadOnlyList<ITemplatePackage>)templatePackages);
+                }
+
                 if (string.IsNullOrEmpty(dn3Path))
                 {
-                    string path = assemblyLocation;
+                    string? path = typeof(Program).Assembly.Location;
```

Comment:
nit: Why change from `assemblyLocation` to `typeof(Program).Assembly.Location` difference is just that we will check one more invalid path at start which is `C:\GIT\templating\artifacts\dotnetnew3.dll\Microsoft.TemplateEngine.sln`.